src/diagram/components/dialog/edit-variable-dialog.tsx

```diff
@@ -1,74 +1,145 @@
+          label="Notes"
+          maxCharacters={kMaxNotesCharacters}
+          rows={2}
+          setValue={variable.setDescription}
+          spellCheck={true}
+          value={variable.description || ""}
+        />
+      </div>
+      <div className={classNames("dialog-content__col dialog-content__col-2", { "is-output-variable": hasInputs })}>
+        <TextAreaRow
+          disabled={true}
+          inputId="evd-expression"
+          label="Expression"
+          value={variable.expression || ""}
+        />
+        <TextAreaRow
+          disabled={true}
+          inputId="evd-calculation"
+          label="Calculation"
+          value={calculation || ""}
+        />
+        <TextAreaRow
+          disabled={hasInputs}
+          inputId="evd-units"
+          label="Unit"
+          value={displayUnit || ""}
+          setValue={variable.setUnit}
+        />
+        <NumberRow
+          disabled={hasInputs}
+          inputId="evd-value"
+          label="Value"
+          realValue={displayValue}
+          setRealValue={variable.setValue}
+        />
+        <div className="dialog-error-messages">
+          { errorMessage && <ErrorMessage valueError={errorMessage} /> }
+        </div>
+      </div>
+      <div className="variable-type-icon">
+        { hasInputs ? <IconOutputCard /> : <IconInputCard /> }
+      </div>
     </div>
   );
 });
 
 // Copies data from copyVariable into variable
 export const updateVariable = (variable: VariableType, copyVariable: VariableType) => {
+  // Because copyVariabe is created from a snapshot, it will not have any of the inputs 
+  // that the original variable has. So we need to save the inputs, apply the snapshot,
+  // and then re-attach the inputs.
+  const inputs = [...variable.inputs];
   applySnapshot(variable, getSnapshot(copyVariable));
+  for (const input of inputs) {
+    variable.addInput(input as Instance<IAnyComplexType>);
+  }
 };
 
 interface IEditVariableDialog {
   onClose: () => void;
-  onSave: (variable: VariableType, copyVariable: VariableType) => void;
   variable: VariableType;
 }
-export const EditVariableDialog = ({ onClose, onSave, variable }: IEditVariableDialog) => {
+
+export const EditVariableDialog = ({ onClose, variable }: IEditVariableDialog) => {
   const variableClone = useMemo(() => Variable.create(getSnapshot(variable)), [variable]);
+  const hasInputs = variable.numberOfInputs > 0;
+  const hasExpression = !!variableClone.expression;
+
+  // Because variableClone was created from a snapshot, we will not be able to access 
+  // computed values related to the original variable's inputs. We need to get those values
+  // from the original variable and either update variableClone or pass the values to the
+  // EditVariableDialogContent component separately.
+  const calculationString = variable.calculationString;
+  const errorMessage = variable?.computedValueError || variable?.computedUnitError || variable?.computedUnitMessage;
```

Comment:
I'm not sure about how I handled this. I'm specifically wondering if it would be better to add some actions to the Variable model so we can set the error message and calculation string that way (like I use the existing `setValue` and `setUnit` below) instead of passing down values through new props I added to the EditVariableDialogContent component.